openbas-api/src/main/java/io/openbas/injectors/email/service/ImapService.java

```diff
@@ -325,6 +326,9 @@ private void retrySyncFolders() throws Exception {
                 Thread.sleep(2000);
             }
         }
+        if(imapStore != null && imapStore.isConnected()) {
+            imapStore.close();
```

Comment:
I am not sure we understand each other on this point.
For me, we can close imapstore only if we can reconnect with it

```
    private void retrySyncFolders() throws Exception {
        for (int i = 0; i < 3; i++) {
            try {
                syncFolders();
                break;
            } catch (MessagingException e) {
                log.warning("Retrying connection..." + e.getMessage());
                Thread.sleep(2000);
                if (i == 2) {
                        imapStore.close();
                }
            }
        }
    }
```
But, we want to have the imapstore opened for all the application life. So I don't know exactly the good solution on this one.